src/ooxml/testcases/org/apache/poi/xssf/usermodel/BaseTestXSSFPivotTable.java

```diff
@@ -296,4 +313,45 @@ public void testPivotTableSheetNamesAreCaseInsensitive() {
         // create a pivot table on a different sheet, case insensitive
         offset.createPivotTable(source, new CellReference("W1"));
     }
+
+
+    /**
+     * Verify that when creating a col label it's  created on the correct column
+     * and the count is increased by one.
+     */
+    @Test
+    public void testAddColLabelToPivotTable() {
+        int columnIndex = 0;
+
+        assertEquals(0, pivotTable.getColLabelColumns().size());
+        
+        pivotTable.addColLabel(columnIndex);
+        CTPivotTableDefinition defintion = pivotTable.getCTPivotTableDefinition();
+
+        assertEquals(defintion.getColFields().getFieldArray(0).getX(), columnIndex);
+        assertEquals(defintion.getColFields().getCount(), 1);
+        assertEquals(1, pivotTable.getColLabelColumns().size());
+        
+        columnIndex = 1;
+        pivotTable.addColLabel(columnIndex);
+        assertEquals(2, pivotTable.getColLabelColumns().size());
+        
+        assertEquals(0, (int)pivotTable.getColLabelColumns().get(0));
+        assertEquals(1, (int)pivotTable.getColLabelColumns().get(1));
+    }
+
+    /**
+     * Verify that it's not possible to create a col label outside of the referenced area.
+     */
+    @Test
+    public void testAddColLabelOutOfRangeThrowsException() {
+        int columnIndex = 5;
+
+        try {
+            pivotTable.addColLabel(columnIndex);
+        } catch(IndexOutOfBoundsException e) {
+            return;
```

Comment:
Using `return` in the middle of a method is against most coding standards
Could you move to fail to the try block? Eg fail("expected an exception to be thrown")

Reply:
I have copied the function testAddRowLabelOutOfRangeThrowsException at this class and have changed the content ... :(
I have to looking for an example to change this function.